tests/common/test_run_as_userid.py

```diff
@@ -0,0 +1,162 @@
+            f"Unable to run as userid '{userid}'. Userid must "
+            + "be a string value between 1 to 8 characters in length.",
+        )
+
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_clear_running_userid(
+        self,
+        call_racf_mock: Mock,
+    ):
+        user_admin = UserAdmin(debug=True, run_as_userid="ESWIFT")
+        user_admin.clear_running_userid()
+        call_racf_mock.side_effect = [
+            TestCommonConstants.TEST_EXTRACT_USER_RESULT_BASE_ONLY_SUCCESS_XML,
+            TestCommonConstants.TEST_ALTER_USER_RESULT_SUCCESS_XML,
+        ]
+        stdout = io.StringIO()
+        with contextlib.redirect_stdout(stdout):
+            user_admin.alter(
+                "squidwrd",
+                traits=TestCommonConstants.TEST_ALTER_USER_REQUEST_TRAITS,
+            )
+        success_log = self.ansi_escape.sub("", stdout.getvalue())
+        self.assertEqual(success_log, TestCommonConstants.TEST_ALTER_USER_SUCCESS_LOG)
+
+    def test_get_running_userid(self):
+        user_admin = UserAdmin(run_as_userid="ESWIFT")
+        running_user = user_admin.get_running_userid()
+        self.assertEqual(running_user, TestCommonConstants.TEST_RUNNING_USERID)
+
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_get_user_access(
+        self,
+        call_racf_mock: Mock,
+    ):
+        precheck_profile_as_squidwrd = (
+            TestCommonConstants.TEST_EXTRACT_RESOURCE_RESULT_PRECHECK_SUCCESS_XML
+        )
+        precheck_profile_as_squidwrd = precheck_profile_as_squidwrd.replace(
+            "<message> 00    ESWIFT          READ              ALTER    NO</message>",
+            "<message> 00    ESWIFT          READ               READ    NO</message>",
+        )
+        resource_admin = ResourceAdmin(debug=True, run_as_userid="ESWIFT")
+        call_racf_mock.return_value = precheck_profile_as_squidwrd
+        stdout = io.StringIO()
+        with contextlib.redirect_stdout(stdout):
+            access = resource_admin.get_user_access(
+                "IRR.IRRSMO00.PRECHECK", "XFACILIT", "squidwrd"
+            )
+        success_log = self.ansi_escape.sub("", stdout.getvalue())
+        self.assertEqual(
+            success_log,
+            TestCommonConstants.TEST_EXTRACT_RESOURCE_PRECHECK_AS_SQUIDWRD_LOG,
+        )
+        self.assertEqual(access, "read")
+
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_get_user_access_raises_userid_error(
+        self,
+        call_racf_mock: Mock,
+    ):
+        userid = "squidwrdtest"
+        precheck_profile_as_squidwrd = (
+            TestCommonConstants.TEST_EXTRACT_RESOURCE_RESULT_PRECHECK_SUCCESS_XML
+        )
+        precheck_profile_as_squidwrd = precheck_profile_as_squidwrd.replace(
+            "<message> 00    ESWIFT          READ              ALTER    NO</message>",
+            "<message> 00    ESWIFT          READ               READ    NO</message>",
+        )
```

Comment:
Do we need mock data here to test `UserIdError`?

Reply:
No, UserIdError doesn't check RACF for anything, it only looks at basic string formatting. It does "evaluate" the passed string without any mocking necessary.

Reply:
Let me rephrase the question. It looks like you are setting up mocks here, but if there are not necessary, then there is no point to setting up the mocks, no?